File: submissions/a-p-i/src/main/java/Mario.java
public class Mario {

    private static int invulnerability = 0;
    private static int coins = 0;

    enum State {
        SUPER, SMALL
    }
    public static void f(String input) {

        String[] commands = input.split("\\s+");

        State state = State.SMALL;
        int life = 3;

        for (String command : commands) {
            if (!command.equals("") && invulnerability > 0) {
                invulnerability--;
            }

            switch (command) {
                case "goomba" :
                case "koopa" :
                case "piranha" :

                if (invulnerability <= 0) {
                    switch (state) {
                        case SMALL :
                            life--;
                            if (life == 0) {
                                System.out.print("GAME OVER");
                                return;
                            }
                            break;
                        case SUPER :
                            state = State.SMALL;
                            break;

                    }
                }
                    break;

                case "Mushroom" :
                    if (state == State.SMALL){
                        state = State.SUPER;
                    }
                    break;
                case "Star" :
                    invulnerability = 3;
                    break;
                case "1Up":
                    life++;
                    break;
                case "Bowser" :
                case "Princess" :
                    System.out.print("WIN");
                    return;

                default:
                    try {
                        coins += Integer.parseInt(command);
                        if (coins >= 100 ) {
                            coins -= 100;
                            life++;
                        }
                    } catch (Exception e) {}

            }

        }

        if (life  > 0)
        System.out.print("???");

    }


}

File: submissions/psychotic-axe/challenge9/mario.go

package main

import (
    "fmt"
    "strings"
    "bytes"
    "io/ioutil"
    "strconv"
    "runtime"
)

func main() {
    var s string = "10 40 koopa goomba 20 30 piranha piranha"
                                 
    f(s)
}

func f(s string) {
	var princess, bowser string
	var coins, life, star int
	end := false
	coins, life, star = 0, 3, 0
	princess, bowser = "Princess", "Bowser"
	var tale[] string = strings.Split(s, " ")
	for index,element := range tale {
		if len(element) > 0 {
			if element == princess || element == bowser{
				fmt.Println("WIN")
				end = true
				break
			} else {
				status := len(element)
				if coin, err := strconv.Atoi(element); err == nil {
					coins += coin
					var newlife int
					newlife = coins / 100
					if newlife >= 1 {
						life += newlife
						coins -= newlife*100
					}
					fmt.Println("||life||"+strconv.Itoa(life)+"||")
					fmt.Println("||coins||"+strconv.Itoa(coins)+"||")
				} else {
					fmt.Println("||status||"+strconv.Itoa(status)+"||")
					switch status {
						case 3:
							fmt.Println("||getlife||")
							life+=1
						case 4:
							fmt.Println("||getstar||")
							star = 2	
						case 5:
							if (star < 0) {
								fmt.Println("||looselife||")
								life-=1
							}
						case 6:
							if (star < 0) {
								fmt.Println("||looselife||")
								life-=1
							}
						case 7:
							if (star < 0) {
								fmt.Println("||looselife||")
								life-=1
							}
						case 8:
							fmt.Println("||getlife||")
							life+=1
					}	
				}
				fmt.Println("||life2||"+strconv.Itoa(life)+"||")
				fmt.Println(strconv.Itoa(index)+element+princess+bowser)
			}
			star--
			if life == 0 {
				fmt.Println("GAME OVER")
				end = true
				break
			}
		}
	}
	if !end {
		fmt.Println("???")
	}
}


func dummyfuncforincludes(){
    dat, err := ioutil.ReadFile("pear")

    if err != nil {
        panic(err)
    }

    fmt.Println(string(dat))

    var buffer bytes.Buffer
    fmt.Println(buffer.String())

    b, err := strconv.ParseBool("true")
    if err != nil {
        panic(err)
    }

    fmt.Println(strconv.FormatBool(b))

    r := []string{}
    strings.Join(r, "")

    fmt.Println(runtime.Compiler, runtime.GOARCH, runtime.GOOS)
    fmt.Println(strconv.IntSize)
}